src/steamship/agents/utils.py

```diff
@@ -18,3 +21,49 @@ def with_llm(llm: LLM, context: Optional[AgentContext] = None) -> AgentContext:
 def get_llm(context: AgentContext, default: Optional[LLM] = None) -> Optional[LLM]:
     """Retrieves the LLM from the provided AgentContext (if it exists)."""
     return context.metadata.get(_LLM_KEY, default)
+
+
+def build_chat_history(
+    default_system_message: str, message_selector: MessageSelector, context: AgentContext
+) -> List[Block]:
+    # system message should have already been created in context, but we double-check for safety
+    if context.chat_history.last_system_message:
```

Comment:
I'm also noticing while doing plugin implementations that we tend to not dedupe between this and "default_system_prompt" in configs, which seems like it could confuse some models.